go/vt/sqlparser/normalizer.go

```diff
@@ -352,24 +353,31 @@ func SQLToBindvar(node SQLNode) *querypb.BindVariable {
 		case DecimalVal:
 			v, err = sqltypes.NewValue(sqltypes.Decimal, node.Bytes())
 		case HexNum:
-			v, err = sqltypes.NewValue(sqltypes.HexNum, node.Bytes())
+			buf := make([]byte, 0, len(node.Bytes()))
+			buf = append(buf, "0x"...)
+			buf = append(buf, bytes.ToUpper(node.Bytes()[2:])...)
+			v, err = sqltypes.NewValue(sqltypes.HexNum, buf)
 		case HexVal:
 			// We parse the `x'7b7d'` string literal into a hex encoded string of `7b7d` in the parser
 			// We need to re-encode it back to the original MySQL query format before passing it on as a bindvar value to MySQL
-			var vbytes []byte
-			vbytes, err = node.encodeHexOrBitValToMySQLQueryFormat()
```

Comment:
Replaced and removed this function with a much simpler approach without a regex.